source/includes/connection-snippets/x509-snippets/cpp-connection.cpp

```diff
@@ -0,0 +1,31 @@
+// begin x509 connection
+#include <iostream>
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/json.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/uri.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/options/client.hpp>
+#include <mongocxx/options/ssl.hpp>
+#include <bsoncxx/builder/basic/kvp.hpp>
+
+using bsoncxx::builder::basic::kvp;
+
+int main(int, char**) {
+  mongocxx::instance inst{};
+
+  mongocxx::options::ssl ssl_opts{};
+  ssl_opts.pem_file("/etc/certs/mongodb/client.pem");
+
+  mongocxx::options::client client_opts{};
+  client_opts.ssl_opts(ssl_opts);
+
+  mongocxx::uri uri("mongodb+srv://<cluster-url>/test?authSource=$external&retryWrites=true&w=majority&authMechanism=MONGODB-X509&tls=true");
```

Comment:
I'd recommend removing "test" from the URI since it is not used as the database to authenticate against for X509, so it serves no purpose. I.e. make it

`mongodb+srv://<cluster-url>/?authSource=$external&retryWrites=true&w=majority&authMechanism=MONGODB-X509&tls=true`

Reply:
Unfortunately, I can't remove `/test` from the uri. Although the spec doesn't require it, some of the drivers (for example, go) cause an error without it. So for this PR the goal is to have the drivers examples as similar as possible. We can remove `/test` from all the snippets eventually but for right now, if one of the drivers requires it, for unification purposes we'll have to keep it for the other drivers as well

Reply:
Gotcha, that's fine with me then.